src/org/thoughtcrime/securesms/notifications/NotificationCenter.java

```diff
@@ -311,16 +316,17 @@ private String getNotificationChannel(NotificationManagerCompat notificationMana
     // add notifications & co.
     // --------------------------------------------------------------------------------------------
 
-    public void addNotification(int chatId, int msgId) {
+    public void addNotification(int accountId, int chatId, int msgId) {
         Util.runOnAnyBackgroundThread(() -> {
 
+            DcContext dcContext = context.dcAccounts.getAccount(accountId);
             DcChat dcChat = dcContext.getChat(chatId);
 
             if (!Prefs.isNotificationsEnabled(context) || dcChat.isMuted()) {
                 return;
             }
 
-            if (chatId == visibleChatId) {
+            if (visibleChat.equals(Pair.create(accountId, chatId))) {
```

Comment:
In case you sometimes make visibleChat `null`, remember to change this to `Util.equals(visibleChat, Pair.create(accountId, chatId))`. In case you also switch to a dedicated class `AccountChatId` instead of Pair, remember to override `equals`.

Reply:
it is never null